tests/flagd/pkg/integration/integration.go

```diff
@@ -15,77 +17,31 @@ type ctxStorageKey struct{}
 // ctxClientKey is the key used to pass the openfeature client across context.Context
 type ctxClientKey struct{}
 
-func firstBooleanEvaluationDetails(ctx context.Context) (openfeature.BooleanEvaluationDetails, error) {
-	store, ok := ctx.Value(ctxStorageKey{}).(map[string]openfeature.BooleanEvaluationDetails)
-	if !ok {
-		return openfeature.BooleanEvaluationDetails{}, errors.New("no flag resolution result")
-	}
-
-	var got openfeature.BooleanEvaluationDetails
-	for _, evalDetails := range store {
-		got = evalDetails
-		break
-	}
-
-	return got, nil
-}
-
-func firstStringEvaluationDetails(ctx context.Context) (openfeature.StringEvaluationDetails, error) {
-	store, ok := ctx.Value(ctxStorageKey{}).(map[string]openfeature.StringEvaluationDetails)
-	if !ok {
-		return openfeature.StringEvaluationDetails{}, errors.New("no flag resolution result")
-	}
-
-	var got openfeature.StringEvaluationDetails
-	for _, evalDetails := range store {
-		got = evalDetails
-		break
-	}
-
-	return got, nil
-}
-
-func firstIntegerEvaluationDetails(ctx context.Context) (openfeature.IntEvaluationDetails, error) {
-	store, ok := ctx.Value(ctxStorageKey{}).(map[string]openfeature.IntEvaluationDetails)
-	if !ok {
-		return openfeature.IntEvaluationDetails{}, errors.New("no flag resolution result")
-	}
-
-	var got openfeature.IntEvaluationDetails
-	for _, evalDetails := range store {
-		got = evalDetails
-		break
-	}
-
-	return got, nil
-}
+func aFlagdProviderIsSet(ctx context.Context) (context.Context, error) {
+	pOptions := []flagd.ProviderOption{flagd.WithPort(8013), flagd.WithoutCache()}
+	pOptions = append(pOptions, providerOptions...)
+	provider := flagd.NewProvider(pOptions...)
+	readyChan := make(chan interface{})
```

Comment:
small nitpick: The most common way in go to implement this kind of signal channels is to use a `chan struct{}` and call `close(readyChan)` to signal readiness 

```suggestion
	readyChan := make(chan struct{})
```

Reply:
Wow thanks! That's enlightening.